src/response/cg.jl

```diff
@@ -0,0 +1,62 @@
+# Implementation of the preconditioned conjugate gradient method, which allows
+# for projection operations, in order to solve linear systems in fixed subspace
+# of the whole space.
+
+"""
+Implementation of the conjugate gradient method which allows for preconditioning
+and projection operations along iterations.
+"""
+function CG!(x, A, b; precon=I, proj=ϕ->ϕ,
```

Comment:
Please use lower-case (I never liked the upper-case LOBPCG either)

Reply:
`proj=identity` (not that it makes a difference, but for clarity)

Add a callback agrument.